### PR TITLE
Add blocking label set in submit queue

### PR DESCRIPTION
**Pull request by kargakis**
We need to ignore pull requests that touch the vendor directory every time we want to land a k8s rebase.

/cc @cjwagner @stevekuznetsov 

@mfojtik fyi

### CODE REVIEW

**Comment by mfojtik** [comment]
🍺 thanks!

**Comment by stevekuznetsov** [comment]
Only comment would be that this label stuff will make the query harder for `tide` in the future, maybe?

File: mungegithub/mungers/submit-queue_test.go [hunk 1]
```diff
@@ -882,6 +883,21 @@ func TestSubmitQueue(t *testing.T) {
 			isMerged:         true,
 		},
 		{
+			name:            "Fail because vendor-update label is required to be missing but exists",
+			pr:              ValidPR(),
+			issue:           AdditionalLabelIssue("vendor-update"),
```
Comment (mfojtik) [inline mungegithub/mungers/submit-queue_test.go]: shouldn't this be "vendor-update-approved" ?

Comment (kargakis) [reply mungegithub/mungers/submit-queue_test.go]: No, the label is `vendor-update`.

**Comment by kargakis** [comment]
> Only comment would be that this label stuff will make the query harder for tide in the future, maybe?

This is already baked into `tide` because of the graphql query supporting requiring missing labels.
https://github.com/kubernetes/test-infra/blob/8aaff58561b0753da6b5a44bd916461957564c28/prow/config.yaml#L25

**Comment by stevekuznetsov** [comment]
Ah, if we have the ability to positively and negatively declare labels we need already in `tide` then this is is

/lgtm

**Pull request merged**